app/controllers/Credentials.scala

```diff
@@ -82,16 +84,10 @@ class Credentials @Inject()(val applicationService: ApplicationService,
     }
   }
 
-  def deleteClientSecretAction(applicationId: String, clientSecretId: String): Action[AnyContent] =
-    canChangeClientSecrets(applicationId) { implicit request =>
-      def deleteClientSecret(clientSecretToDelete: String): Future[Result] = {
-        applicationService.deleteClientSecrets(applicationId, request.user.email, Seq(clientSecretToDelete))
-          .map(_ => Redirect(controllers.routes.Credentials.clientSecrets(applicationId)))
-      }
-
-      applicationService.fetchCredentials(applicationId).flatMap { tokens =>
-        tokens.clientSecrets.find(_.id == clientSecretId)
-          .fold(successful(NotFound(errorHandler.notFoundTemplate)))(secret => deleteClientSecret(secret.secret))
-      }
+  def deleteClientSecretAction(appId: String, clientSecretId: String): Action[AnyContent] =
```

Comment:
Should we not be typing appId to UUID here ?  Then we can use a binder that does the fromString and reports a BadRequest where it's not so rather than it throwing an IllegalArgException and returning a probable 500???

Reply:
Had a quick look - applicationId is a String pretty much everywhere in this service (including in the Application case class). I'll change it on the Controller.

Definitely agree that we should be using stronger types, but to do a thorough job will require a fairly major refactor.